test/fixtures/parser/bigquery/select_mixture_of_array_literals.yml

```diff
@@ -0,0 +1,82 @@
+file:
+  statement:
+    select_statement:
+      select_clause:
+      - keyword: SELECT
+      - select_clause_element:
+          expression:
+            array_literal:
+              start_square_bracket: '['
+              end_square_bracket: ']'
+      - comma: ','
+      - select_clause_element:
+          expression:
+            array_literal:
+              start_square_bracket: '['
+              expression:
+                literal: 'false'
+              end_square_bracket: ']'
+      - comma: ','
+      - select_clause_element:
+          expression:
+          - keyword: ARRAY
+          - comparison_operator: <
+          - data_type_identifier: BOOLEAN
+          - comparison_operator: '>'
+          - array_literal:
+              start_square_bracket: '['
+              expression:
+                literal: 'false'
+              end_square_bracket: ']'
+      - comma: ','
+      - select_clause_element:
+          expression:
+            array_literal:
+              start_square_bracket: '['
+              expression:
+                literal: "'a'"
+              end_square_bracket: ']'
+          alias_expression:
+            keyword: AS
+            identifier: strcol1
+      - comma: ','
+      - select_clause_element:
+          expression:
+          - keyword: ARRAY
+          - comparison_operator: <
+          - data_type_identifier: string
```

Comment:
Looks to me as though the angle brackets are parsed as less-than and greater-than operators still?

Reply:
IIRC, we chose not to add a new token type, thinking that was tidier. Do you think we should?

Reply:
(In other words, this was a decision several months back when we first added support for angle brackets, not a recent decision.)

Let me know what you think -- I could see this current behavior causing some linter rules to get confused.

Reply:
@NiallRees: I'll go ahead and change this, i.e. change it so the angle brackets parse output uses a new token. Definitely easier to understand the parse output this way.